examples/safe_enable_fdcan.rs

```diff
@@ -0,0 +1,35 @@
+#![deny(warnings)]
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+
+extern crate panic_itm;
+
+use cortex_m_rt::entry;
+use stm32h7xx_hal::rcc::{rec, ResetEnable};
+use stm32h7xx_hal::{pac, prelude::*};
+
+fn enable_fdcan(rec: rec::Fdcan) {
+    rec.enable().kernel_clk_mux(rec::FdcanClkSel::PLL1_Q);
+
+    // rec is dropped here, and can never be changed again
+}
+
+#[entry]
+fn main() -> ! {
+    let dp = pac::Peripherals::take().unwrap();
+
+    // Constrain and Freeze power
+    let pwr = dp.PWR.constrain();
+    let vos = pwr.freeze();
+
+    // Constrain and Freeze clock
+    let rcc = dp.RCC.constrain();
+    let ccdr = rcc.sys_ck(100.mhz()).freeze(vos, &dp.SYSCFG);
+
+    enable_fdcan(ccdr.peripheral.FDCAN);
```

Comment:
This example doesn't seem exactly useful to me like....it shows absolutely nothing to the user unless they read the correct bits from the proper registers, can we maybe make this an actual example?

Or is it more meant like an example on how to handle peripherals with the new approach?

Reply:
It is a demonstration, indeed it doesn't actually do anything. It can be removed if it doesn't add anything to the example folder